### PR TITLE
Fix incorrect context name being created when using new tanzu platform endpoints

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it

Fix incorrect context name being created when using new endpoints

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

**Before this change:**
```
$ tz login --staging --endpoint https://platform-dev.tanzu.broadcom.com

$ tz context list
  NAME                                      ISACTIVE  TYPE   PROJECT  SPACE
  tpsm-5a2f0150                             false     tanzu
```

**After this change:**
```
$ tz login --staging --endpoint https://platform-dev.tanzu.broadcom.com

$ tz context list
  NAME                                      ISACTIVE  TYPE   PROJECT  SPACE
  Falcons-staging-5a2f0150                  true      tanzu
```

### Release note

```release-note
Fix incorrect context name being created when using new tanzu platform endpoints
```

### CODE REVIEW

**Review by marckhouzam — APPROVED** [review]
I actually hit this problem yesterday while debugging and it was pretty annoying (and wrong).
Thanks for fixing!

In the PR description did you mean to refer to #801 instead of #808 (which is the current PR 😄 )

File: pkg/command/login_test.go [hunk 1]
```diff
@@ -11,61 +11,91 @@ import (
...
-			forceCSP:  true,
+			orgName:       "MyOrg",
+			ucpEndpoint:   "https://ucp.custom-endpoint.com",
+			endpoint:      "https://custom-endpoint.com",
+			saasEndpoints: []string{centralconfig.DefaultTanzuPlatformEndpoint},
+			isStaging:     true,
+			expected:      "MyOrg-staging-70217fc3",
+			forceCSP:      true,
+		},
+		// Test case for normal input with new ucpEndpoint and platform endpoints which is actually a SaaS endpoints
```
Comment (vuil) [inline pkg/command/login_test.go:60]: nit: which is actually a SaaS endpoints -> which are actually SaaS endpoints. Same for L68

**Review by vuil — APPROVED** [review]
Thanks for the fix and updating the tests to clarify the behavior.
Just a small nit on the comments.

File: pkg/command/login_test.go (hunk 1 above)
Comment (anujc25) [reply pkg/command/login_test.go]: Thanks. Updated.

**Pull request merged**